lib/route.js

```diff
@@ -307,12 +307,28 @@ function buildRouting (options) {
         constraints.version = opts.version
       }
 
-      const headHandler = router.find('HEAD', opts.url, constraints)
-      const hasHEADHandler = headHandler != null
+      let headHandler = null
+      let hasHEADHandler = false
+
+      if (shouldExposeHead && opts.method === 'GET') {
+        // We're using the "router.on" method in order to understand if there is an identical route
+        // in router, only with "HEAD" method.
+        // once "has" method will be implemented on 'find-my-way' side, we will use it.
+        try {
+          router.on('HEAD', opts.url, { constraints }, routeHandler, context)
+          router.off('HEAD', opts.url, constraints)
+        } catch (error) {
```

Comment:
can we fix this in find-my-way properly? feels super bad to add this.

Reply:
> can we fix this in find-my-way properly? feels super bad to add this.

Please take a look at the discussion at #4986. There's a detailed explanation of how and why this is the only way atm.
Also, we discussed it here.